### PR TITLE
Refactor the client

### PR DESCRIPTION
**Pull request by skejserjensen**
This PR refactors the client to match the conventions used in the server and adds doc comments.

### CODE REVIEW

File: client/src/main.rs [hunk 1]
```diff
@@ -12,205 +12,271 @@
...
-                    Err(message) => eprintln!("{}", message),
-                };
-            }
-            Err(message) => eprintln!("error: cannot connect to {} due to a {}", address, message),
-        }
+    let (maybe_host, maybe_query_file_path) = parse_command_line_arguments(args);
+
+    // Create the Tokio runtime.
+    let runtime = Runtime::new()
+        .map_err(|error| format!("error: unable to initialize run-time due to {}", error))?;
```
Comment (CGodiksen) [inline client/src/main.rs:77]: It does not seem like we use the `error:` prefix anywhere in the server. Maybe remove this.

File: client/src/main.rs [hunk 2]
```diff
@@ -12,205 +12,271 @@
...
+
+    // Connect to the server.
+    let host = maybe_host.unwrap_or_else(|| DEFAULT_HOST.to_owned());
+    let flight_service_client = connect(&runtime, &host, DEFAULT_PORT).map_err(|error| {
+        format!(
+            "error: cannot connect to {} due to a {}",
```
Comment (CGodiksen) [inline client/src/main.rs:83]: It does not seem like we use the `error:` prefix anywhere in the server. Maybe remove this.

File: client/src/main.rs [hunk 3]
```diff
@@ -12,205 +12,271 @@
...
+    })?;
+
+    // Execute the queries.
+    if let Some(query_file) = maybe_query_file_path {
+        file(runtime, flight_service_client, &query_file)
     } else {
-        eprintln!("error: unable to initialize run-time");
+        repl(runtime, flight_service_client)
     }
+    .map_err(|error| format!("error: cannot execute queries due to {}", error))
```
Comment (CGodiksen) [inline client/src/main.rs:95]: It does not seem like we use the `error:` prefix anywhere in the server. Maybe remove this.

File: client/src/main.rs [hunk 4]
```diff
@@ -12,205 +12,271 @@
...
     if let Some(mut home) = dirs::home_dir() {
         home.push(history_file_name);
         let _ = editor.append_history(&home);
     }
+
     Ok(())
 }
 
+/// Execute an action, a command, or a query. Returns [`Error`] if the action, command, or query
+/// could not be executed and their result retrieved.
```
Comment (CGodiksen) [inline client/src/main.rs:207]: Should this be `...or their result could not be retrieved.`?

**Review by CGodiksen — CHANGES_REQUESTED** [review]

File: client/src/main.rs (hunk 1 above)
Comment (skejserjensen) [reply client/src/main.rs:77]: Fixed.

File: client/src/main.rs (hunk 2 above)
Comment (skejserjensen) [reply client/src/main.rs:83]: Fixed.

File: client/src/main.rs (hunk 3 above)
Comment (skejserjensen) [reply client/src/main.rs:95]: Fixed.

File: client/src/main.rs (hunk 4 above)
Comment (skejserjensen) [reply client/src/main.rs:207]: Fixed.

**Review by CGodiksen — APPROVED** [review]

**Review by chrthomsen — APPROVED** [review]

**Pull request merged**